### DISCUSSION TITLE
Mangiatoia per pesci con Esp32 e Stepper Motor

### DISCUSSION THREAD

**Discussion by SimoneNafra** [indomus/forum#3091] [Nuove idee, proposte, progetti e guide]
Buonasera,

sto cercando di realizzare una mangiatoia automatica da integrare con HA.
Sto utilizzando un wemos d1 mini pro integrato su HA con Esphome
un motore stepper 28byj48 e un driver uln2003.

Ho integrato tutto seguendo la guida https://esphome.io/components/stepper/index.html# e riesco a muovere correttamente il motore.

Vorrei adesso creare un automazione per muoverlo ad un determinato orario...
Con la configurazione che ho inserito su esphome ho solamente l'azione set_value che non mi permette di assegnare un valore incrementale.

L'idea in pseudocodice potrebbe essere riassunta come:

if (time = 9:00 or 12:00 or 16:00){
motor_position = motor_position + 50
}
Come posso implementarlo in HA?
Aggiungo che nella guida è presente il paragrafo "stepper.report_position Action" che credo faccia al caso mio ma non riesco a capire come funziona e dove devo inserire quel codice.

Nel dettaglio i miei dubbi sono:

on_...: (cosa significa on_...??)
  then:
  - stepper.report_position:
      id: my_stepper
      position: 250    (perchè un "get" position ha un valore associato?)
   
Grazie in anticipo a chi mi potrà aiutare.

Saluti

**Comment by neonjollyroma** [comment]
io uso questa comandando poi il giro di quanti passi serve a me (292) con un automazione homeassistant

esphome:
  name: stepperfeeder
  friendly_name: Stepper Feeder
 
esp8266:
  board: d1_mini
 
# Abilita log, OTA, API, Wi-Fi
logger:
# Enable Home Assistant API

api:
  encryption:
    key: "........"

ota:
  - platform: esphome
    password: "........."

wifi:
  networks:
  - ssid: "wifi"
    password: "passwifi"    
  #- ssid: "wifi2"
  #  password: "passwifi2"
  power_save_mode: none  
  ap:
    ssid: "Feeder Fallback"
    password: "passwififallback"


web_server:
  port: 80

captive_portal:
  
# Si:  D1, D2, D5, D6, D7
# No: D3, D4 o D8
 
# Stepper motore 28BYJ-48 con driver ULN2003
stepper:
  - platform: uln2003
    id: my_stepper
    pin_a: D1
    pin_b: D2
    pin_c: D6
    pin_d: D5
    max_speed: 250
    step_mode: HALF_STEP
    sleep_when_done: true
 
button:
  - platform: template
    name: "Dai cibo"
    id: manual_step_button
    on_press:
      then:
        - lambda: |-
            int target = id(my_stepper).current_position + 292;
            id(my_stepper).set_target(target);
